cmd/manager/main.go

```diff
@@ -146,12 +146,17 @@ func main() {
 
 // addMetrics will create the Services and Service Monitors to allow the operator export the metrics by using
 // the Prometheus operator
-func addMetrics(ctx context.Context, cfg *rest.Config, namespace string) {
-	if err := serveCRMetrics(cfg); err != nil {
+func addMetrics(ctx context.Context, cfg *rest.Config) {
+	// Get the namespace the operator is currently deployed in.
+	operatorNs, err := k8sutil.GetOperatorNamespace()
+	if err != nil {
 		if errors.Is(err, k8sutil.ErrRunLocal) {
 			log.Info("Skipping CR metrics server creation; not running in a cluster.")
 			return
 		}
+	}
+
+	if err := serveCRMetrics(cfg, operatorNs); err != nil {
```

Comment:
Old code, we were returning an error, in this case, instead of proceeding. don't we need it now? Also, do we need to change the error message in line 154?

Reply:
The changes to the addMetrics function are directly suggested by the Operator SDK upgrade guide as part of the transition to v0.16.x
https://sdk.operatorframework.io/docs/upgrading-sdk-version/version-upgrade-guide/#v016x
Related PRs:
https://github.com/operator-framework/operator-sdk/pull/2606
https://github.com/operator-framework/operator-sdk/pull/2603
https://github.com/operator-framework/operator-sdk/pull/2601